core/commands/destroy_command.rb

```diff
@@ -59,10 +66,56 @@ def show_help
 
 After running the vagrant destroy this command also deletes the
 libvirt and VirtualBox boxes using low-level commands.
-HELP
+    HELP
     @ui.out(info)
   end
 
+  # Method gets the libvirt virtual machines names list.
+  #
+  # @return [Array] virtual machines names.
+  def libvirt_vm_list
+    check_command('virsh list --name --all',
+                  'Unable to get Libvirt vm\'s list')[:output].split("\n")
+  end
+
+  # Method gets the VirtualBox virtual machines names list.
+  #
+  # @return [Array] virtual machines names.
+  def virtualbox_vm_list
+    check_command('VBoxManage list vms | grep -o \'"[^\"]*"\' | tr -d \'"\'',
+                  'Unable to get VirtualBox vm\'s list')[:output].split("\n")
+  end
+
+  # Method gets the AWS instances names list.
+  #
+  # @return [Array] instances names list.
+  def aws_vm_list
+    @aws_instance_ids.map { |instance| instance[:node_name] }
+  end
+
+  # Print virtual machines names list of all providers.
+  def show_vm_list
+    vm_list = libvirt_vm_list + virtualbox_vm_list + aws_vm_list
+    @ui.info("Virtual machines list: #{vm_list}")
+  end
+
+  # Destroy all virtual machines of all providers that correspond with the node_name.
+  #
+  # @param node_name [String] regexp of the node name.
+  def destroy_machine_by_name(node_name)
+    node_name_regexp = Regexp.new(node_name)
+    vm_list = { libvirt: libvirt_vm_list, virtualbox: virtualbox_vm_list, aws: aws_vm_list }
+    vm_list.each do |provider, nodes|
+      vm_list[provider] = nodes.select { |node| node =~ node_name_regexp }
+    end
+    @ui.info("Virtual machines to destroy: #{vm_list.values.flatten}")
+    @ui.info('Do you want to continue? [y/n]')
+    return if gets[0] != 'y'
```

Comment:
User may enter 'Y' or ' y'. It should be still correct input. Please remove all starting and trailing spaces from the user input and force input to the lower case.